dlt/extract/utils.py

```diff
@@ -119,6 +131,38 @@ def check_compat_transformer(name: str, f: AnyFun, sig: inspect.Signature) -> in
     return meta_arg
 
 
+def wrap_async_generator(
+    gen: AsyncGenerator[TDataItems, None]
+) -> Generator[Awaitable[TDataItems], None, None]:
+    """Wraps an async generator into a list of awaitables"""
+    exhausted = False
+    busy = False
```

Comment:
using an async.lock here will fail on python 3.8, and 3.9 because there is no current loop on the main thread. i am not sure why it is allowed in python 3.10. threading locks should not be used with coroutines as far as I understand.